internal/artifacts/manager.go

```diff
@@ -216,14 +206,53 @@ func (m *Manager) GetOfficialExtensions(ctx context.Context, versionString strin
 	return extensions, nil
 }
 
-// GetExtensionImage pulls and exports an extension image.
+// GetInstallerImage pulls and stoers in OCI layout installer image.
+func (m *Manager) GetInstallerImage(ctx context.Context, arch Arch, versionString string) (string, error) {
+	version, err := semver.ParseTolerant(versionString)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse version: %w", err)
+	}
+
+	if version.LT(m.options.MinVersion) {
+		return "", fmt.Errorf("version %s is not supported, minimum is %s", version, m.options.MinVersion)
+	}
+
+	tag := "v" + version.String()
+
+	ociPath := filepath.Join(m.storagePath, string(arch)+"-installer-"+tag)
+
+	// check if already fetched
+	if _, err := os.Stat(ociPath); err != nil {
+		resultCh := m.sf.DoChan(ociPath, func() (any, error) {
+			return nil, m.fetchInstallerImage(arch, tag, ociPath)
+		})
+
+		select {
+		case <-ctx.Done():
+			return "", ctx.Err()
+		case result := <-resultCh:
+			if result.Err != nil {
+				return "", result.Err
+			}
+		}
+	}
+
+	return ociPath, nil
+}
+
+// GetInstallerImageRef returns the installer image reference for the given version.
+// func (m *Manager) GetInstallerImageRef(versionString string) string {
+// 	return m.imageRegistry.Repo(InstallerImage).Tag("v" + versionString).String()
+// }
```

Comment:
Is keeping this commented block intentinal?

Reply:
nope... thanks, I forgot to GC it